src/azul/indexer.py

```diff
@@ -183,3 +183,72 @@ def merge(cur_bundles, new_bundles):
         for bundle in cur_bundles_by_id.values():
             assert bundles.setdefault(bundle['uuid'], bundle) is bundle
         return list(bundles.values())
+
+    def delete(self, dss_notification: Mapping[str, Any]) -> None:
+        bundle_uuid = dss_notification['match']['bundle_uuid']
+        bundle_version = dss_notification['match']['bundle_version']
+        index = dss_notification['match']['index']
+
+        es_client = self.properties.elastic_search_client
+
+        search_query = {
+            "query": {
+                "match": {
+                    "bundles.uuid.keyword": bundle_uuid
+                }
+            }
+        }
+
+        current_doc = es_client.search(index=index,
+                                       doc_type="doc",
+                                       body=search_query)
+
+        hits = current_doc['hits']['hits']
+
+        def update_call(cur_doc_id, body, version):
+            es_client.update(index=index,
+                             doc_type="doc",
+                             id=cur_doc_id,
+                             body=body,
+                             version=version
+                             )
+
+        def on_success():
+            response_to_return = es_client.search(index=index,
+                                                  doc_type="doc",
+                                                  body=search_query)
+
+            return response_to_return
+
+        def on_conflict(cur_doc_id, body, error):
+            log.warning("Deletion of doc %s failed with a conflict error %s . Retrying", current_doc_id, error)
+            hit_doc_updated = es_client.get(index=index,
+                                            doc_type="doc",
+                                            id=cur_doc_id)
+
+            update_call(current_doc_id, body, hit_doc_updated['_version'])
+
+        for hit in hits:
+            current_doc_id = hit['_id']
+            hit_doc = es_client.get(index=index,
+                                    doc_type="doc",
+                                    id=current_doc_id)
+
+            current_hit_doc = hit_doc['_source']
+            for bundle in current_hit_doc['bundles']:
+                if bundle['version'] == bundle_version:
```

Comment:
Don't you also need to compare the bundle UUID? Two distinct bundles (different UUID) could have the same version.